### ISSUE TITLE
Upload a YAML or JSON file 

### ISSUE THREAD

**Issue by guoh1988** [kubernetes/dashboard#2037]
##### Environment

```
Dashboard version:gcr.io/google_containers/kubernetes-dashboard-amd64   v1.6.1 
Kubernetes version:Client Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.2", GitCommit:"a55267932d501b9fbd6d73e5ded47d79b5763ce5", GitTreeState:"clean", BuildDate:"2017-04-14T13:36:25Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.2", GitCommit:"a55267932d501b9fbd6d73e5ded47d79b5763ce5", GitTreeState:"clean", BuildDate:"2017-04-14T13:36:25Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
Operating system:CentOS Linux release 7.2.1511
Node.js version:NA
Go version:go version go1.6.3 linux/amd64
```

##### Steps to reproduce

1.etcd
/usr/bin/etcd --name=default --data-dir=/var/lib/etcd/default.etcd --listen-client-urls=http://0.0.0.0:2379
2.flannel
/usr/bin/flanneld -etcd-endpoints=http://192.168.0.111:2379 -etcd-prefix=/kube/network
3.kube-apiserver 
/usr/bin/kube-apiserver --logtostderr=true --v=0 --etcd-servers=http://192.168.0.111:2379 --insecure-bind-address=0.0.0.0 --allow-privileged=false --service-cluster-ip-range=172.17.0.0/16 --admission-control=NamespaceLifecycle,NamespaceExists,LimitRanger,SecurityContextDeny,ResourceQuota --advertise-address=192.168.0.111
4.kube-controller-manager
/usr/bin/kube-controller-manager --logtostderr=true --v=0 --master=http://192.168.0.111:8080
5.kube-scheduler 
/usr/bin/kube-scheduler --logtostderr=true --v=0 --master=http://192.168.0.111:8080
6.kubelet
/usr/bin/kubelet --logtostderr=true --v=0 --api-servers=http://192.168.0.111:8080 --address=0.0.0.0 --port=10250 --hostname-override=192.168.0.111 --allow-privileged=false --pod-infra-container-image=registry.access.redhat.com/rhel7/pod-infrastructure:latest
7.kube-proxy
/usr/bin/kube-proxy --logtostderr=true --v=0 --master=http://192.168.0.111:8080
8.dashboard-1.6.1
[root@allserver docker]# kubectl create -f kubernetes-dashboard-no-rbac.yaml 
deployment "kubernetes-dashboard" created
service "kubernetes-dashboard" created
logs 
Using HTTP port: 8443
Using apiserver-host location: http://192.168.0.111:8080
Creating API server client for http://192.168.0.111:8080
Successful initial request to the apiserver, version: v1.5.2
Creating in-cluster Heapster client
Using random key for csrf signing

9.Login http://192.168.0.111:8080/ui， Deploy a Containerized APP BY Upload a YAML or JSON file 
**error message**
error validating "kubernetes-dashboard-no-rbac.yaml": error validating data: Get http://localhost:8080/swaggerapi/apis/extensions/v1beta1: dial tcp [::1]:8080: getsockopt: connection refused; if you choose to ignore these errors, turn validation off with --validate=false 

Deploying file has failed

unable to connect to a server to handle "deployments": Get http://localhost:8080/api: dial tcp [::1]:8080: getsockopt: connection refuse

**Comment by floreks** [comment]
Fix for this bug also fixes #2041. Right now deploy app from file will work only for in-cluster configs. We have to fix that.

**Comment by floreks** [comment]
Fixed by #2046.
